### DISCUSSION TITLE
How to Integrate and build Opendds in Android Open Source Project (AOSP)

### DISCUSSION THREAD

**Discussion by mayur-desh** [OpenDDS/OpenDDS#3532] [Q&A] [answered]
How can we integrate,build and develop applications for android in Android Open Source Project (AOSP).

**Comment by jwillemsen** [comment, marked as answer]
See https://github.com/objectcomputing/OpenDDS/blob/master/docs/android.md

**Reply by jmmorato** [reply]
https://github.com/iguessthislldo/OpenDDS-Android looks promising as well ;)

**Reply by iguessthislldo** [reply]
I'd recommend android.md over the scripts because I don't always keep those scripts up to date and they're aimed more at testing out various build settings for OpenDDS than a tool for end users to build OpenDDS.

The [smartlock demo](https://github.com/oci-labs/opendds-smart-lock) demo is an another resource and just got an update so it should work with the latest of everything.

**Reply by mitza-oci** [reply]
@mayur-desh 
> How can we integrate, build and develop applications for android in Android Open Source Project (AOSP).

Do the responses on this thread answer your question?  Is there anything different about building for AOSP vs. Google's Android distributions?
